### PR TITLE
chore: add user classes, fake data, user tests remove enzyme

### PR DESCRIPTION
**Pull request by elanjm12**
Remove enzyme from all user tests. Replace with rtl. Add user classes in typescript and use faker js to populate fake data.

### CODE REVIEW

File: src/js/users/components/__tests__/Create.test.jsx [hunk 1]
```diff
@@ -22,80 +51,46 @@ describe("<CreateUser />", () => {
...
-            errorPassword: "Password too short"
-        });
-        expect(wrapper).toMatchSnapshot();
+    it("should render correct error messages", () => {
+        renderWithRouter(<CreateUser {...props} />, state, history);
+        const saveButton = screen.getByRole("button", { name: "Save" });
+        saveButton.click();
+        expect(screen.getByText("Please specify a username")).toBeInTheDocument();
+        expect(screen.getByText("Passwords must contain at least 8 characters")).toBeInTheDocument();
+        expect(props.onCreate).not.toHaveBeenCalled();
```
Comment (igboyes) [inline src/js/users/components/__tests__/Create.test.jsx]: Example of making test more readable.
```suggestion
        renderWithRouter(<CreateUser {...props} />, state, history);
        
        userEvent.click(screen.getByRole("button", { name: "Save" }));
        
        expect(screen.getByText("Please specify a username")).toBeInTheDocument();
        expect(screen.getByText("Passwords must contain at least 8 characters")).toBeInTheDocument();
        expect(props.onCreate).not.toHaveBeenCalled();
```

File: src/js/users/components/__tests__/Create.test.jsx [hunk 2]
```diff
@@ -22,80 +51,46 @@ describe("<CreateUser />", () => {
...
+    it("should not display error messages and should call onCreate when valid info entered", () => {
+        renderWithRouter(<CreateUser {...props} />, state, history);
+        const userNameInput = screen.getByLabelText("username");
+        const passwordInput = screen.getByLabelText("password");
+        const saveButton = screen.getByRole("button", { name: "Save" });
+        fireEvent.change(userNameInput, { target: { value: "newUsername" } });
+        fireEvent.change(passwordInput, { target: { value: "newPassword" } });
+        expect(props.onCreate).not.toHaveBeenCalled();
+        saveButton.click();
+        expect(props.onCreate).toHaveBeenCalled();
```
Comment (igboyes) [inline src/js/users/components/__tests__/Create.test.jsx]: Example of making test more readable.

```suggestion
        renderWithRouter(<CreateUser {...props} />, state, history);
        
        fireEvent.change(screen.getByLabelText("username"), { target: { value: "newUsername" } });
        fireEvent.change(screen.getByLabelText("password"), { target: { value: "newPassword" } });
        
        expect(props.onCreate).not.toHaveBeenCalled();
        
        screen.getByRole("button", { name: "Save" }).click();
        
        expect(props.onCreate).toHaveBeenCalled();
```

File: src/js/users/classes.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    last_password_change: string;
+    permissions: Permissions;
+    primary_group: Group;
+}
+
+export function getFakeUserDetail(): UserDetail {
+    const groups = [];
+    times(5, (i: number) =>
+        groups.push({ id: faker.random.alphaNumeric(9, { casing: "lower" }), name: `group${i}`, toggled: true })
+    );
```
Comment (igboyes) [inline src/js/users/classes.ts]: Based on the Lodash docs, this should work.
```suggestion
    const groups = times(5, (i: number) => ({
        id: faker.random.alphaNumeric(9, { casing: "lower" }),
        name: `group${i}`,
        toggled: true
    }));
```

**Review by igboyes — CHANGES_REQUESTED** [review]
Thanks for getting this process started.

Please make these tests more readable by:
1. Introducing logical whitespace.
    * Put prep, events, and assertion in separate blocks. Examples provided.
    * If you do mulitple rounds and events and assertions, split them up with whitespace.
2. Removing unnecessary variables that are only used once.

DRY-ify your unnecessarily repetitious `createAppStore()` function and associated setup. This looks like it has essentially been copy-pasted between test modules.

Use `userEvent` wherever possible instead of `fireEvent`. Google for benefits of `user-events`.

You have added functions with prefixes `getFakeDocuments`. The `get` prefix implies you are getting a property from an object. Use `create` instead.

**Review by ReeceHoffmann — DISMISSED** [review]

**Review by igboyes — DISMISSED** [review]

**Pull request merged**